coderd/database/queries/users.sql

```diff
@@ -117,7 +117,9 @@ RETURNING *;
 UPDATE
 	users
 SET
-	hashed_password = $2
+	hashed_password = $2,
+	hashed_one_time_passcode = NULL,
+	one_time_passcode_expires_at = NULL
```

Comment:
Let me know your thoughts: Would placing OTPs on a separate table be less dangerous? Separation is safer to me, there are fewer updates if somebody maliciously requests new OTP codes.

@DanielleMaywood @mafredri

Reply:
I'm not sure I understand what the danger would be with having the OTPs on the same table.

Reply:
Thinking loud, feel free to challenge

By separating OTPs into a different table, we limit the number of updates to sensitive fields in the user table. In the future, this would also make it easier to apply specific security measures to OTPs alone - such as faster expiration times, restricted access, or even separate encryption rules.
It’s not necessarily a large risk, but keeping high-frequency, high-sensitivity data in its own isolated space could add another layer of safety.

Reply:
I can see some benefits with a separate table, like limiting the RBAC scope. We don't need to grab system-level permission to modify the users table (OTP request), we can have a narrowly scoped permission that's limited to the OTP table. Although, I'm not sure how much that adds in terms of security since we'll need system to read a user anyway.

Separate table also opens up the ability to have multiple reset tokens in-flight (although not saying we should support the use-case).

Smaller table also makes queries faster, reduces writes and simplifies cleanup (delete where token expired vs update, truncate, etc.).

Reply:
@mtojek and I discussed this and we've decided that for now this is okay and if needed we can change it in the future.